certbot/eff.py

```diff
@@ -0,0 +1,95 @@
+"""Subscribes users to the EFF newsletter."""
+import logging
+
+import requests
+import zope.component
+
+from certbot import constants
+from certbot import interfaces
+
+
+logger = logging.getLogger(__name__)
+
+
+def handle_subscription(config):
+    """High level function to take care of EFF newsletter subscriptions.
+
+    The user may be asked if they want to sign up for the newsletter if
+    they have not already specified.
+
+    :param .IConfig config: Client configuration.
+
+    """
+    if config.email is None:
+        if config.eff_email:
+            _report_failure("you didn't provide an e-mail address")
+        return
+    if config.eff_email is None:
+        config.eff_email = _want_subscription()
+    if config.eff_email:
+        subscribe(config.email)
+
+
+def _want_subscription():
+    """Does the user want to be subscribed to the EFF newsletter?
+
+    :returns: True if we should subscribe the user, otherwise, False
+    :rtype: bool
+
+    """
+    prompt = (
+        'Would you be willing to share your email address with the '
+        "Electronic Frontier Foundation, a founding partner of the Let's "
+        'Encrypt project and the non-profit organization that develops '
+        "Cerbot? We'd like to send you email about our work to encrypt "
+        'the web and protect its users.')
```

Comment:
did you come up with this wording - or did someone like peter or kurt have input?